backend/dataall/modules/datasets/tasks/dataset_subscription_task.py

```diff
@@ -163,17 +161,6 @@ def publish_sns_message(
                             f'Failed to deliver message {message} due to: {e}'
                         )
 
-    @staticmethod
-    def sns_call(message, environment):
-        aws_session = SessionHelper.remote_session(environment.AwsAccountId)
-        sns = aws_session.client('sns', region_name=environment.region)
-        response = sns.publish(
-            TopicArn=f'arn:aws:sns:{environment.region}:{environment.AwsAccountId}:{environment.subscriptionsConsumersTopicName}',
```

Comment:
In SnsDatasetClient - which you used to replace this method - the TopicArn differs, there it's:
`f'arn:aws:sns:{environment.region}:{environment.AwsAccountId}:{environment.subscriptionsProducersTopicName}'`

Reply:
Nice catch! Updated the topic